exe/3scale-update

```diff
@@ -0,0 +1,215 @@
+
+    missing_application_plans.each do |plan|
+      plan.delete('links')
+      plan.delete('default') # TODO: handle default plan§
+
+      if plan.delete('custom') # TODO: what to do with custom plans?
+        puts "skipping custom plan #{plan}"
+      else
+        client.create_application_plan(service_target_id, plan)
+      end
+    end
+
+    application_plan_mapping = client.list_service_application_plans(service_target_id).map do |plan_target|
+      plan = plans.find{|plan| plan.fetch('system_name') == plan_target.fetch('system_name') }
+
+      [plan['id'], plan_target['id']]
+    end
+
+
+    metrics_mapping = client.list_metrics(service_target_id).map do |target|
+      metric = metrics.find{|metric| metric.fetch('system_name') == target.fetch('system_name') }
+      metric ||= {}
+
+      [metric['id'], target['id']]
+    end.to_h
+
+    mapping_rules = source_client.list_mapping_rules(service_id)
+    mapping_rules_target = client.list_mapping_rules(service_target_id)
+
+    puts "the source service has #{mapping_rules.size} mapping rules"
+    puts "the target has #{mapping_rules_target.size} mapping rules"
+
+    unique_mapping_rules_target = mapping_rules_target.dup
+
+    missing_mapping_rules = mapping_rules.reject do |mapping_rule|
+      matching_metric = unique_mapping_rules_target.find do |target|
+        compare_hashes(mapping_rule, target, %w(pattern http_method delta)) &&
+            metrics_mapping.fetch(mapping_rule.fetch('metric_id')) == target.fetch('metric_id')
+      end
+
+      unique_mapping_rules_target.delete(matching_metric)
+    end
+
+    puts "missing #{missing_mapping_rules.size} mapping rules"
+
+    missing_mapping_rules.each do |mapping_rule|
+      mapping_rule.delete('links')
+      mapping_rule['metric_id'] = metrics_mapping.fetch(mapping_rule.delete('metric_id'))
+      client.create_mapping_rule(service_target_id, mapping_rule)
+    end
+    puts "created #{missing_mapping_rules.size} mapping rules"
+
+    puts "extra #{unique_mapping_rules_target.size} mapping rules"
+    puts unique_mapping_rules_target.each{|rule| rule.delete('links') }
+
+    application_plan_mapping.each do |source_id, target_id|
+      limits = source_client.list_application_plan_limits(source_id)
+      limits_target = client.list_application_plan_limits(target_id)
+
+      missing_limits = limits.reject { |limit| limits_target.find{|limit_target| limit.fetch('period') == limit_target.fetch('period') } }
+
+      missing_limits.each do |limit|
+        limit.delete('links')
+        client.create_application_plan_limit(target_id, metrics_mapping.fetch(limit.fetch('metric_id')), limit)
+      end
+      puts "target application plan #{target_id} is missing #{missing_limits.size} from the source plan #{source_id}"
+    end
+  when 'help'
+    print_help.call
+  when nil
+    print_help.call("missing subcommand")
+  else
+    print_help.call("unknown command #{command}")
```

Comment:
probably should `exit 1` after to return proper exit code, the same missing subcommand